uber/models.py

```diff
@@ -588,6 +608,75 @@ def lookup_attendee(self, first_name, last_name, email, zip_code):
 
             raise ValueError('attendee not found')
 
+        def add_promo_code_to_attendee(self, attendee, code):
+            """
+            Convenience method for adding a promo code to an attendee.
+
+            This method sets both the `promo_code` and `promo_code_id`
+            properties of `attendee`. Due to the way the `Attendee.promo_code`
+            relationship is defined, the `Attendee.promo_code_id` isn't
+            automatically set, which makes this method a nice way of setting
+            both.
+
+            Arguments:
+                attendee (Attendee): The Attendee for which the promo code
+                    should be added.
+                code (str): The promo code as typed by an end user, or an
+                    empty string to unset the promo code.
+
+            Returns:
+                str: Either a failure message or an empty string
+                    indicating success.
+            """
+            code = code.strip()
```

Comment:
This would error out, however it's an error to assign `None` to code, so I think that's okay. I am not opposed to adding an additional check for `None` and other falsey values though, so I'll go ahead and do that.